app/src/main/java/com/codesoom/assignment/controllers/TaskController.java

```diff
@@ -0,0 +1,79 @@
+package com.codesoom.assignment.controllers;
+
+// 1. Read Collection - GET /tasks ==> 완료
+// 2. Read Item - GET /tasks/{id} ==> 완료
+// 3. Create - POST /tasks ==> 완료
+// 4. update - PUT/PATCH /tasks{id} ==> 완료
+// 5. delete - DELETE /tasks{id}
+
+import com.codesoom.assignment.NotFoundException;
+import com.codesoom.assignment.models.Task;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.http.HttpStatus;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@RestController
+@RequestMapping("/tasks")
+public class TaskController {
+
+    private List<Task> tasks = new ArrayList<>();
+    private Long newId = 0L;
+    NotFoundException notFoundException = new NotFoundException();
```

Comment:
미리 생성해두지 않고 필요할 때에만 생성할 수 있도록 수정하는 것이 좋아 보입니다.

Reply:
1) 이 프로그램에서는 id를 무조건 포함하여 할 일을 생성해야 하는데도 미리 생성해 두지 않아야 할까요?

2) 아래와 같이 수정해 보았는데 트레이너님 피드백에 맞는 수정인지 확인 부탁 드립니다!

```suggestion
    public class TaskRepository {

    public Map<Long, Task> taskStore = new HashMap<>();
    AtomicLong idCounter = new AtomicLong(0);

    public Long generateId() {
        return idCounter.getAndIncrement();
    }

}
```

Reply:
아 제 이야기는 예외 이야기였습니다 ㅎㅎ 예외는 던질 때에만 생성해 주세요.

Reply:
아하 ㅎㅎ 예외에 대해 말씀하신거였군요!

```suggestion
        public Task findTask(Long id) {
        if (taskRepository.taskStore.get(id) == null) {
            throw new NotFoundTaskIdException();
        }
        return taskRepository.taskStore.get(id);
        }
```
위와 같이 던질때 생성하는 것으로 수정해보았습니다!